services/src/main/java/org/keycloak/services/resources/KeycloakApplication.java

```diff
@@ -316,15 +315,14 @@ public void importRealm(RealmRepresentation rep, String from) {
                     RealmModel realm = manager.importRealm(rep);
                     ServicesLogger.LOGGER.importedRealm(realm.getName(), from);
                 }
-                session.getTransactionManager().commit();
             } catch (Throwable t) {
-                session.getTransactionManager().rollback();
-                if (!exists) {
-                    ServicesLogger.LOGGER.unableToImportRealm(t, rep.getRealm(), from);
-                }
+                session.getTransactionManager().setRollbackOnly();
+                throw t;
+            }
+        } catch (Throwable t) {
+            if (!exists) {
+                ServicesLogger.LOGGER.unableToImportRealm(t, rep.getRealm(), from);
             }
```

Comment:
Does this really always work? Previously the commit was done prior to this catch, but now it's done by auto close, which happens after this catch doesn't it?